packages/controllers/src/snaps/json-schemas/snap-manifest.schema.json

```diff
@@ -0,0 +1,125 @@
+{
+  "title": "Snap Manifest",
+  "description": "The Snap manifest file MUST be named `snap.manifest.json` and located in the package root directory.",
+  "type": "object",
+  "additionalProperties": false,
+  "required": [
+    "version",
+    "description",
+    "proposedName",
+    "source",
+    "initialPermissions",
+    "manifestVersion"
+  ],
+  "properties": {
+    "version": {
+      "type": "string",
+      "title": "Version",
+      "description": "MUST be a valid SemVer version string and equal to the corresponding `package.json` field.",
+      "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
+    },
+    "description": {
+      "type": "string",
+      "title": "Description",
+      "description": "MUST be a non-empty string less than or equal to 280 characters. A short description of the Snap.",
+      "minLength": 1,
+      "maxLength": 280
+    },
+    "proposedName": {
+      "type": "string",
+      "title": "Proposed Name",
+      "description": "MUST be a string less than or equal to 214 characters. The Snap author's proposed name for the Snap. The Snap host application may display this name unmodified in its user interface. The proposed name SHOULD be human-readable.",
+      "minLength": 1,
+      "maxLength": 214,
+      "pattern": "^(?:[A-Za-z0-9-_]+( [A-Za-z0-9-_]+)*)|(?:(?:@[A-Za-z0-9-*~][A-Za-z0-9-*._~]*/)?[A-Za-z0-9-~][A-Za-z0-9-._~]*)$"
+    },
+    "repository": {
+      "title": "Repository",
+      "description": "MAY be omitted. If present, MUST be equal to the corresponding package.json field.",
+      "oneOf": [
+        {
+          "type": "null"
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["type", "url"],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["git"]
+            },
+            "url": {
+              "type": "string",
+              "minLength": 1
+            }
+          }
+        }
+      ]
+    },
+    "source": {
+      "type": "object",
+      "title": "Source",
+      "description": "Specifies some Snap metadata and where to fetch the Snap during installation.",
+      "additionalProperties": false,
+      "required": ["shasum", "location"],
+      "properties": {
+        "shasum": {
+          "type": "string",
+          "description": "MUST be the Base64-encoded string representation of the SHA-256 hash of the Snap source file.",
+          "minLength": 44,
+          "maxLength": 44,
+          "pattern": "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$"
+        },
+        "location": {
+          "title": "Source Location",
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["npm"],
```

Comment:
should `local` be allowed here

Reply:
We decided that we will maybe do this in the future.